### PR TITLE
Update hive caching documentation

### PR DESCRIPTION
**Pull request by mosabua**
Addressing the late feedback on the PR ..

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 1]
```diff
@@ -61,12 +61,16 @@ systems.
...
-size and are well suited for ORC or Parquet format objects.
+Caching can operate in two modes. The default async mode provides the queried
+data directly and caches any objects asynchronously afterwards. Any following
+queries requesting the cached objects are served directly from the cache.
+
+The second mode is a read-through cache. After first retrieval from storage by
+any query, objects are cached in the local cache storage on the workers. Objects
+are cached on local storage of each worker and managed by a BookKeeper
+component. Workers can request cached objects from other workers to avoid
+requests from the object storage. The cache chunks are 1MB in size and are well
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:72]: `The cache chunks are 1MB in size and are well...` this is valid for both async and read-through mode

Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst:72]: ok .. I will separate the paragraph

Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst]: Separated the different concerns into paragraphs.

**Comment by mosabua** [comment]
Can you review and merge @electrum ?

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 2]
```diff
@@ -61,12 +61,19 @@ systems.
...
+queries requesting the cached objects are served directly from the cache.
+
+The second mode is a read-through cache. After first retrieval from storage by
+any query, objects are cached in the local cache storage on the workers.
+
+In both modes objects are cached on local storage of each worker and managed by
+a BookKeeper component. Workers can request cached objects from other workers to
+avoid requests from the object storage.
+
+The cache chunks are 1MB in size and are well suited for ORC or Parquet format
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:75]: "format objects" -> file formats

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 3]
```diff
@@ -61,12 +61,19 @@ systems.
...
-retrieval from storage by any query, objects are cached in the local cache
-storage on the workers. Objects are cached on local storage of each worker and
-managed by a BookKeeper component. Workers can request cached objects from other
-workers to avoid requests from the object storage. The cache chunks are 1MB in
-size and are well suited for ORC or Parquet format objects.
+Caching can operate in two modes. The default async mode provides the queried
+data directly and caches any objects asynchronously afterwards. Any following
+queries requesting the cached objects are served directly from the cache.
+
+The second mode is a read-through cache. After first retrieval from storage by
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:68]: second -> other

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 4]
```diff
@@ -137,7 +143,8 @@ In all cases, you should avoid using the root partition and disk of the node and
 instead attach at multiple dedicated storage devices for the cache on each node.
 The cache uses the drive up to a configurable percentage. Storage should be
 local on each coordinator and worker node. The directory needs to exist before
-Presto starts.
+Presto starts. We recommend using two directories to improve performance of the
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:146]: Why two, specifically? Is this two directories per disk, or is it trying to say that multiple disks can improve performance? What if you only have one cache disk, should you still create two directories on it?

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 3 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst:68]: Fixed

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 2 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst:75]: fixed

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 4 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst:146]: Add clarification

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 5]
```diff
@@ -137,7 +143,8 @@ In all cases, you should avoid using the root partition and disk of the node and
 instead attach at multiple dedicated storage devices for the cache on each node.
 The cache uses the drive up to a configurable percentage. Storage should be
 local on each coordinator and worker node. The directory needs to exist before
-Presto starts.
+Presto starts. We recommend using at least two directories, located on separate
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:146]: How about

> We recommend using multiple devices to improve performance of the cache.

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 6]
```diff
@@ -137,7 +143,8 @@ In all cases, you should avoid using the root partition and disk of the node and
 instead attach at multiple dedicated storage devices for the cache on each node.
 The cache uses the drive up to a configurable percentage. Storage should be
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:144]: drive -> disk

**Pull request merged**